app/env.py

```diff
@@ -31,3 +31,33 @@
 
 # comma-separated list:
 ALLOWED_HOSTS = os.environ.get("ALLOWED_HOSTS", "").split(",")
+
+###
+# credentials encryption secrets and parameters
+#
+# to rotate credentials encryption secrets (and settings):
+# 1. update environment:
+#    - set GRAVYVALET_ENCRYPT_SECRET to a new, long, random string (...no commas, tho)
+#    - add the old secret to GRAVYVALET_ENCRYPT_SECRET_PRIORS (comma-separated list)
+#    - (optional) update key-derivation parameters with best practices du jour
+# 2. call `.rotate_encryption()` on every `ExternalCredentials` (perhaps via celery task
+#    `addon_service.credentials.tasks.schedule_encryption_rotation__celery`)
+# 3. remove the old secret from GRAVYVALET_ENCRYPT_SECRET_PRIORS
+GRAVYVALET_ENCRYPT_SECRET = os.environ.get("GRAVYVALET_ENCRYPT_SECRET")
+GRAVYVALET_ENCRYPT_SECRET_PRIORS = tuple(
+    filter(bool, os.environ.get("GRAVYVALET_ENCRYPT_SECRET_PRIORS", "").split(","))
+)
+# optional overrides for scrypt key derivation parameters (when unset or "0", use sensible defaults)
+# see https://datatracker.ietf.org/doc/html/rfc7914#section-2
+GRAVYVALET_SALT_BYTE_COUNT = int(os.environ.get("GRAVYVALET_SALT_BYTE_COUNT", 0))
```

Comment:
Are any of these, aside from `GRAVYVALET_ENCRYPT_SECRET`, required to be set in the configs?

Also, why default to `0` here but override in the app code, rather than just defaulting to non-zero here?

Reply:
none required besides `GRAVYVALET_ENCRYPT_SECRET` (and that's optional too when `DEBUG`)

yeah, now looking again the `0` defaults here are weird and inconsistent... have added actual defaults here (and also replaced `SCRYPT_COST` with `SCRYPT_COST_LOG2` so you don't ever have to bother computing 2^18)